### PR TITLE
Introduce `glob` built-ins

### LINKED ISSUE #965: Add glob built-in
I would like to propose the addition of glob functionality. With that I mean the possibility of matching simple strings against patters like `foo.*`, `foo:*`, `foo:*:bar`. The current functionality `reg.globs_match` does not implement that. Happy to tackle a PR!

This type of pattern matching is usually desirable when matching URNs, particularly URLs, for example matching a particular subdomain `*.mydomain.com`, `{foo,bar,baz}.mydomain.com`.

A library that implements this is [glob](https://github.com/gobwas/glob).

As always, happy to provide a PR!

### PR DESCRIPTION
**Pull request by aeneasr**
This patch introduces built-ins `glob.match` and `glob.quote_meta`.

Closes #965

### CODE REVIEW

**Comment by aeneasr** [comment]
Wow, that's a lot of vendor updates, sorry about that - it's the result of `vendor update`. IMO this project should switch to go modules and no longer rely on `vendor`. It would be trivial for me to introduce a vulnerability in those changed 55k+ lines of code with no way of actually reviewing it.

**Comment by srenatus** [comment]
AFAIK `dep` will prune vendored sources, and only include what's needed. Could be worth trying, replacing glide...

**Comment by aeneasr** [comment]
`go mod` is the way to go, switched all of my projects over. Some things have issues with it though like `goimports` and `go-swagger` but other than that it's really great!

File: docs/book/language-reference.md [hunk 1]
```diff
@@ -88,6 +88,40 @@ complex types.
...
+| ``glob.match("[a-c]at", [], "lat", output)`` | ``false`` | A glob with character-range matchers. |
+| ``glob.match("[!a-c]at", [], "cat", output)`` | ``false`` | A glob with negated character-range matchers. |
+| ``glob.match("[!a-c]at", [], "lat", output)`` | ``true`` | A glob with negated character-range matchers. |
+| ``glob.match(""{cat,bat,[fr]at}", [], "cat", output)`` | ``true`` | A glob with pattern-alternatives matchers. |
+| ``glob.match(""{cat,bat,[fr]at}", [], "bat", output)`` | ``true`` | A glob with pattern-alternatives matchers. |
+| ``glob.match(""{cat,bat,[fr]at}", [], "rat", output)`` | ``true`` | A glob with pattern-alternatives matchers. |
+| ``glob.match(""{cat,bat,[fr]at}", [], "at", output)`` | ``false`` | A glob with pattern-alternatives matchers. |
+
+
+Exactly one of ``cert`` and ``secret`` must be present.
```
Comment (tsandall) [inline docs/book/language-reference.md]: Looks like this got copied accidentally.

**Comment by aeneasr** [comment]
Oh yeah, my bad - removed that!

**Comment by tsandall** [comment]
Hrmmm. Looks like the vendoring update was lost. Once that's fixed we can merge this.

**Comment by aeneasr** [comment]
Whoopsie, fixed!

**Comment by aeneasr** [comment]
🙏

**Pull request merged**